src/main/java/io/confluent/connect/elasticsearch/Validator.java

```diff
@@ -238,7 +272,91 @@ private void validateSsl() {
     }
   }
 
+  private void validateConnection() {
+    try {
+      if (!client.ping(RequestOptions.DEFAULT)) {
+        throw new IOException("Cannot connect to Elasticsearch.");
+      }
+    } catch (IOException e) {
+      String errorMessage = String.format(
+          "Could not connect to Elasticsearch due to %s",
+          e.getMessage()
+      );
+      addErrorMessage(CONNECTION_URL_CONFIG, errorMessage);
+
+      if (config.isAuthenticatedConnection()) {
+        errorMessage = String.format(
+            "Could not authenticate the user. Check the '%s' and '%s'. Error message: %s.",
+            CONNECTION_USERNAME_CONFIG,
+            CONNECTION_PASSWORD_CONFIG,
+            e.getMessage()
+        );
+        addErrorMessage(CONNECTION_USERNAME_CONFIG, errorMessage);
+        addErrorMessage(CONNECTION_PASSWORD_CONFIG, errorMessage);
+      }
+
+      if (config.isSslEnabled()) {
+        errorMessage = String.format(
+            "Could not connect to Elasticsearch. Check your SSL settings. Error message: %s",
+            e.getMessage()
+        );
+
+        addErrorMessage(SECURITY_PROTOCOL_CONFIG, errorMessage);
+      }
+
+      if (config.isKerberosEnabled()) {
+        errorMessage = String.format(
+            "Could not connect to Elasticsearch. Check your Kerberos settings. Error message: %s",
+            e.getMessage()
+        );
+
+        addErrorMessage(KERBEROS_PRINCIPAL_CONFIG, errorMessage);
+        addErrorMessage(KERBEROS_KEYTAB_PATH_CONFIG, errorMessage);
+      }
+
+      if (config.isBasicProxyConfigured()) {
+        errorMessage = String.format(
+            "Could not connect to Elasticsearch. Check your proxy settings. Error message: %s",
+            e.getMessage()
+        );
+        addErrorMessage(PROXY_HOST_CONFIG, errorMessage);
+        addErrorMessage(PROXY_PORT_CONFIG, errorMessage);
+
+        if (config.isProxyWithAuthenticationConfigured()) {
+          addErrorMessage(PROXY_USERNAME_CONFIG, errorMessage);
+          addErrorMessage(PROXY_PASSWORD_CONFIG, errorMessage);
+        }
+      }
+    }
+  }
+
   private void addErrorMessage(String property, String error) {
     values.get(property).addErrorMessage(error);
   }
+
+  private RestHighLevelClient createClient() {
+    ConfigCallbackHandler configCallbackHandler = new ConfigCallbackHandler(config);
+    return new RestHighLevelClient(
+        RestClient
+            .builder(
+                config.connectionUrls()
+                    .stream()
+                    .map(HttpHost::create)
+                    .collect(Collectors.toList())
+                    .toArray(new HttpHost[config.connectionUrls().size()])
+            )
+            .setHttpClientConfigCallback(configCallbackHandler)
+            .setRequestConfigCallback(configCallbackHandler)
+    );
+  }
+
+  private boolean hasErrors() {
```

Comment:
Don't we have a similar existing method already? If not, can be a good candidate to be generalized and moved to a common library.

Reply:
that would have to be added in AK
another option would be to add to internal libraries but that wouldn't be available to any OSS connectors